### ISSUE TITLE
[QUERY] The issuer certificate does not have an appropriate value for the Basic Constraints extensions

### ISSUE THREAD

**Issue by shahriak** [Azure/azure-sdk-for-net#27187]
### Library name and version

Azure.Security.KeyVault.Certificates 4.2.0

### Query/Question

Hello,
I created the following self-signed certificate directly in the Azure KeyVault. I am trying to mint an intermediate certificate from this using the code shown below, however, I have been hitting this error "Encountered an unhandled exception
      System.ArgumentException: The issuer certificate does not have an appropriate value for the Basic Constraints extension. (Parameter 'issuerCertificate')"

![tds1](https://user-images.githubusercontent.com/12322998/155591827-fdfe1863-892d-493b-9dea-518f6e7d9032.PNG)
![tdsb](https://user-images.githubusercontent.com/12322998/155591831-5bb4f36a-44e5-478d-a324-bd67a2c66353.PNG)
![tdsc](https://user-images.githubusercontent.com/12322998/155591833-25d2cdd6-b4e2-41f7-88e9-4adbbe915392.PNG)
![tdsd](https://user-images.githubusercontent.com/12322998/155593592-e11c450f-f2c6-4b0e-9d80-3760bf2851bc.PNG)

Code snippet:
```
        public virtual X509Certificate2 CreateIntermediateCertificateAuthority(string subjectName, int keySize, DateTimeOffset expirationDate, X509Certificate2 issuer, List<string> subjectAlternativeNameDnsNames = null, List<Uri> subjectAlternativeNameUris = null)
        {
            using RSA rsa = RSA.Create(keySize);
            CertificateRequest certificateRequest = new CertificateRequest($"L = {subjectName}", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);

            certificateRequest.CertificateExtensions.Add(
                new X509BasicConstraintsExtension(true, true, 1, true));

            // subject key identifier must NOT be marked critical: https://www.ietf.org/rfc/rfc3280.txt
            certificateRequest.CertificateExtensions.Add(
                new X509SubjectKeyIdentifierExtension(certificateRequest.PublicKey, false));

            SubjectAlternativeNameBuilder sanBuilder = new SubjectAlternativeNameBuilder();
            bool addSan = false;

            if (subjectAlternativeNameDnsNames != null)
            {
                foreach (string dnsName in subjectAlternativeNameDnsNames)
                {
                    sanBuilder.AddDnsName(dnsName);
                }

                addSan = true;
            }

            if (subjectAlternativeNameUris != null)
            {
                foreach (Uri uri in subjectAlternativeNameUris)
                {
                    sanBuilder.AddUri(uri);
                }

                addSan = true;
            }

            if (addSan)
            {
                certificateRequest.CertificateExtensions.Add(sanBuilder.Build());
            }

            string allowedNames = subjectName.Trim().TrimStart('.');
            certificateRequest.CertificateExtensions.Add(NameConstraintsExtensionHelper.GetNameConstraintX509Extension(allowedNames));

            certificateRequest.CertificateExtensions.Add(
                new X509KeyUsageExtension(
                    X509KeyUsageFlags.DigitalSignature |
                    X509KeyUsageFlags.NonRepudiation |
                    X509KeyUsageFlags.KeyEncipherment |
                    X509KeyUsageFlags.KeyCertSign,
                    true));

            byte[] serialBytes = new byte[4];
            this.random.NextBytes(serialBytes);
            expirationDate = expirationDate.AddDays(-1);
            X509Certificate2 certificate = certificateRequest.Create(issuer, NotBefore, expirationDate, serialBytes);
            return certificate.CopyWithPrivateKey(rsa);
        }

```

### Environment

IDE and version: Visual Sudio 16.11.4

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by heaths** [comment]
Duplicate of Azure/azure-rest-api-specs#11962

The service needs to support this before we can in the SDK. I will also ping the service team internally about the referenced issue, but recommend subscribing to that one.
